scripts/maintenance/computational-clusters/autoscaled_monitor/cli.py

```diff
@@ -0,0 +1,154 @@
+import asyncio
+from pathlib import Path
+from typing import Annotated
+
+import parse
+import rich
+import typer
+from dotenv import dotenv_values
+
+from . import core as api
+from .constants import (
+    DEFAULT_COMPUTATIONAL_EC2_FORMAT,
+    DEFAULT_DYNAMIC_EC2_FORMAT,
+    DEPLOY_SSH_KEY_PARSER,
+)
+from .ec2 import autoscaling_ec2_client, cluster_keeper_ec2_client
+from .models import AppState
+
+state: AppState = AppState(
+    dynamic_parser=parse.compile(DEFAULT_DYNAMIC_EC2_FORMAT),
+    computational_parser=parse.compile(DEFAULT_COMPUTATIONAL_EC2_FORMAT),
+)
+
+app = typer.Typer()
+
+
+def _parse_environment(deploy_config: Path) -> dict[str, str | None]:
+    repo_config = deploy_config / "repo.config"
+    assert repo_config.exists()
+    environment = dotenv_values(repo_config)
+    if environment["AUTOSCALING_EC2_ACCESS_KEY_ID"] == "":
+        rich.print(
+            "Terraform variables detected, looking for repo.config.frozen as alternative."
+            " TIP: you are responsible for them being up to date!!"
+        )
+        repo_config = deploy_config / "repo.config.frozen"
+        assert repo_config.exists()
+        environment = dotenv_values(repo_config)
+
+        if environment["AUTOSCALING_EC2_ACCESS_KEY_ID"] == "":
+            error_msg = (
+                "Terraform is necessary in order to check into that deployment!\n"
+                f"install terraform (check README.md in {state.deploy_config} for instructions)"
+                "then run make repo.config.frozen, then re-run this code"
+            )
+            rich.print(error_msg)
+            raise typer.Abort(error_msg)
+    assert environment
+    return environment
+
+
+@app.callback()
+def main(
+    deploy_config: Annotated[
+        Path, typer.Option(help="path to the deploy configuration")
+    ]
+):
+    """Manages external clusters"""
+
+    state.deploy_config = deploy_config.expanduser()
+    assert (
+        deploy_config.is_dir()
+    ), "deploy-config argument is not pointing to a directory!"
+    state.environment = _parse_environment(deploy_config)
+
+    # connect to ec2s
+    state.ec2_resource_autoscaling = autoscaling_ec2_client(state)
+    state.ec2_resource_clusters_keeper = cluster_keeper_ec2_client(state)
+
+    assert state.environment["EC2_INSTANCES_KEY_NAME"]
+    state.dynamic_parser = parse.compile(
+        f"{state.environment['EC2_INSTANCES_NAME_PREFIX']}-{{key_name}}"
+    )
+    if state.environment["CLUSTERS_KEEPER_EC2_INSTANCES_PREFIX"]:
+        state.computational_parser = parse.compile(
+            f"{state.environment['CLUSTERS_KEEPER_EC2_INSTANCES_PREFIX']}-{DEFAULT_COMPUTATIONAL_EC2_FORMAT}"
+        )
+
+    # locate ssh key path
+    for file_path in deploy_config.glob("**/*.pem"):
+        # very bad HACK
+        if (
+            any(_ in f"{file_path}" for _ in ("sim4life.io", "osparc-master"))
+            and "openssh" not in f"{file_path}"
```

Comment:
it would may not work for tip science. Feel free to check the key names for it in `osparc-config` repo

Reply:
adapted to skip license.